### DISCUSSION TITLE
21. HTTPS에서 대칭키와 비대칭키를 모두 사용하는 이유는 무엇인가요?

### DISCUSSION THREAD

**Discussion by eun-byeol** [jewelry-box/cs-interview#23] [General]
HTTPS에서 대칭키와 비대칭키를 모두 사용하는 이유는 무엇인가요?

**Comment by linirini** [comment]
### 대칭키와 비대칭키의 비교
특성 | 대칭키 암호화 | 비대칭키 암호화
-- | -- | --
키의 개수 | 하나의 키로 암호화와 복호화 수행 | 공개키와 개인키 두 개의 키 사용
속도 | 빠름 | 느림
보안성 | 키 관리에 따라 보안성이 달라짐 | 키 분리가 되어 키 교환이 안전함
사용 용도 | 데이터 전송, 파일 암호화 등 | 인증, 키 교환, 디지털 서명
키 관리 | 키를 송신자와 수신자 모두 안전하게 공유해야 함 | 공개키는 자유롭게 공유, 개인키는 비밀 유지
### 하이브리드 암호화
대칭키는 속도는 빠르지만, 보안 상 취약점(키 공유)이 존재한다.
비대칭키는 보안 상 더 안전하지만, 속도가 느리다.
따라서, HTTPS에서는 **키를 공유하는 과정에서 비대칭키 암호화 방식**을 사용하여 안전하게 키를 공유한 후, 실제 **데이터 송수신 과정에선 대칭키 암호화 방식**을 사용하여 속도를 최적화하였다.
